spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/CachingConnectionFactory.java

```diff
@@ -801,6 +801,9 @@ private void logicalClose(ChannelProxy proxy) throws Exception {
 			if (this.target != null && !this.target.isOpen()) {
 				synchronized (targetMonitor) {
 					if (this.target != null && !this.target.isOpen()) {
+						if (this.target instanceof PublisherCallbackChannel) {
+							this.target.close(); // emit nacks if necessary
```

Comment:
M-m-m. From one side I have asked about the backport for exactly these 3 lines.
But from other we always try to keep each change covered with test, especially if it is a bug.
Like this?

Anyway you have came here somehow to find this and fix